website_and_docs/content/documentation/webdriver/waits.en.md

```diff
@@ -1,495 +1,206 @@
 ---
-title: "Waits"
+title: "Waiting Strategies"
 linkTitle: "Waits"
 weight: 6
 aliases: ["/documentation/en/webdriver/waits/"]
 ---
 
-WebDriver can generally be said to have a blocking API.
-Because it is an out-of-process library that
-_instructs_ the browser what to do,
-and because the web platform has an intrinsically asynchronous nature,
-WebDriver does not track the active, real-time state of the DOM.
-This comes with some challenges that we will discuss here.
-
-From experience,
-most intermittent issues that arise from use of Selenium and WebDriver
-are connected to _race conditions_ that occur between
-the browser and the user's instructions.
-An example could be that the user instructs the browser to navigate to a page,
-then gets a **no such element** error
-when trying to find an element.
-
-Consider the following document:
-
-```html
-<!doctype html>
-<meta charset=utf-8>
-<title>Race Condition Example</title>
-
-<script>
-  var initialised = false;
-  window.addEventListener("load", function() {
-    var newElement = document.createElement("p");
-    newElement.textContent = "Hello from JavaScript!";
-    document.body.appendChild(newElement);
-    initialised = true;
-  });
-</script>
-```
-
-The WebDriver instructions might look innocent enough:
-
-{{< tabpane langEqualsHeader=true >}}
+Perhaps the most common challenge for browser automation is ensuring
+that the web application is in a state to execute a particular
+Selenium command as desired. the processes often end up in
+a _race condition_ where sometimes the browser gets into the right
+state first (things work as intended) and sometimes the Selenium code
+executes first (things do not work as intended). This is the
+primary cause of _flaky tests_.
+
+All navigation commands wait for a specific `readyState` value
+based on the [page load strategy]({{< ref "drivers/options#pageloadstrategy" >}}) (the 
+default is `"complete"`) before the driver returns control to the code.
+The `readyState` only concerns itself with loading assets defined in the HTML, 
+but loaded JavaScript assets often result in changes to the site,
+and elements you need to interact with may not yet be on the page
+when the code is ready to execute the next Selenium command.
+
+Similarly, in a lot of single page applications, elements get dynamically
+added to a page or change visibility based on a click. For example, the box
+element takes a second to show up after clicking on the "adder" button, so
+even trying to locate that button will error without some form of synchronization: 
+
+{{< tabpane text=true langEqualsHeader=true >}}
   {{< tab header="Java" >}}
-driver.get("file:///race_condition.html");
-WebElement element = driver.findElement(By.tagName("p"));
-assertEquals(element.getText(), "Hello from JavaScript!");
+{{< gh-codeblock path="examples/java/src/test/java/dev/selenium/waits/WaitsTest.java#L23-L28" >}}
```

Comment:
I would rewrite the test without the 'assert for exception', since the purpose of the test is not to make sure the assertion is thrown but for the element interaction to be possible. Below in a section there is a reference that says this code would not fail using a wait, but it's the other way around.